action.sh

```diff
@@ -42,41 +42,41 @@ jq --slurp '
   | reverse
   | .[]
     | .original
-  ' </tmp/versions.txt >/tmp/sorted.txt
+  ')
 
 echo "::group::All available K3s versions (newest on top)"
-cat /tmp/sorted.txt
+echo "$versions_sorted"
 echo "::endgroup::"
 
 # The "latest" version is not directly exposed, but we hard-code its meaning.
 if [[ "${VERSION}" == "latest" ]]; then
-  VERSION=$(head -n 1 /tmp/sorted.txt | jq -r)
+  VERSION=$(jq --slurp <<< "$versions_sorted" --raw-output '.[0]')
```

Comment:
These could also be written as [`first`](https://stedolan.github.io/jq/manual/v1.5/#first,last,nth(n)) if you find that more readable.
```suggestion
  VERSION=$(jq --slurp <<< "$versions_sorted" --raw-output 'first')
```

Reply:
This is fine with `.[0]` — less magic is better.